src/api/app/controllers/webui/comments_controller.rb

```diff
@@ -67,13 +72,12 @@ def permitted_params
     params.require(:comment).permit(:body, :parent_id)
   end
 
-  def find_commentable
-    commentable = [Project, Package, BsRequest].find { |klass| klass.name == params[:commentable_type] }
+  def set_commented
+    @commentable_type = [Project, Package, BsRequest].find { |klass| klass.name == params[:commentable_type] }
+    return if @commentable_type
 
-    if commentable.nil?
-      redirect_to(root_path, error: 'Failed to create comment')
-    else
-      @commented = commentable.find(params[:commentable_id])
-    end
+    @commented = @commentable_type.find_by(id: params[:commentable_id])
```

Comment:
Any reason for not doing  `@commentable_type.find(params[:commentable_id])`?

Reply:
No, :thinking: I'll change it

Reply:
Wait, that code is wrong... I need to rewrite it...